tests/core/nogo/nolint/nolint_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nolint_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel_testing"
+)
+
+func TestMain(m *testing.M) {
+	bazel_testing.TestMain(m, bazel_testing.Args{
+		Main: `
+-- BUILD.bazel --
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_tool_library", "nogo")
+
+nogo(
+    name = "nogo",
+    vet = True,
+    visibility = ["//visibility:public"],
+)
+
+go_library(
+    name = "has_errors",
+    srcs = ["has_errors.go"],
+    importpath = "haserrors",
+)
+
+-- has_errors.go --
+package haserrors
+
+//nolint:buildtag
+// +build build_tags_error
+
+import (
+	"fmt"
+	"sync/atomic"
+)
+
+func F() {}
+
+func Foo() bool {
+	x := uint64(1)
+	_ = atomic.AddUint64(&x, 1)
+	if F == nil { //nolint:all
+		return false
+	}
+	fmt.Printf("%b", "hi", true || true) //nolint:printf,bools
+	return true || true //nolint
+}
+
+func InlineComment() {
+	s := "hello" // nolint
+	fmt.Printf("%d", s)
+}
+
+func LinterMatch() bool {
+	return true || true //nolint:printf
+}
+`,
```

Comment:
I would much prefer if you could split this test into smaller tests, each have 1-2 `//nolint` directive demonstrating different behaviors.

Reply:
Will do